src/Data/Repositories/WalletRepository.cs

```diff
@@ -385,5 +386,128 @@ public async Task<(bool, string?)> FinaliseWallet(Wallet selectedWalletToFinalis
+                        IsCompromised = false,
+                        IsFinalised = true,
+                        WalletAddressType = WalletAddressType.NativeSegwit,
+                        IsHotWallet = false,
+                        IsBIP39Imported = false,
+                        ImportedOutputDescriptor = outputDescriptor,
+                        BIP39Seedphrase = null,
+                        InternalWalletSubDerivationPath = null,
+                        InternalWalletMasterFingerprint = null,
+                        Keys = keys,
+                        InternalWalletId = null,
+                        MofN = 1,
+
+
+                    };
+                }
+                else if (strategyBase is P2WSHDerivationStrategy p2WshDerivationStrategy && p2WshDerivationStrategy.Inner is MultisigDerivationStrategy multisigDerivationStrategy)
+                {
+                    
+                    wallet = new Wallet
+                    {
+                        Id = 0,
+                        CreationDatetime = DateTimeOffset.UtcNow,
+                        Name = name,
+                        Description = description,
+                        IsArchived = false,
+                        IsCompromised = false,
+                        IsFinalised = true,
+                        WalletAddressType = WalletAddressType.NativeSegwit,
+                        IsHotWallet = false,
+                        IsBIP39Imported = false,
+                        ImportedOutputDescriptor = outputDescriptor,
+                        BIP39Seedphrase = null,
+                        Keys = keys,
+                        MofN = multisigDerivationStrategy.RequiredSignatures
+                    };
+                }
+                else
+                {
+                    _logger.LogError("Invalid output descriptor");
+                    return (false, "Invalid output descriptor");
+                    
+                }
+                
+                //Persist wallet
+                var addResult = await AddAsync(wallet);
+                if (addResult.Item1 == false)
+                {
+                    _logger.LogError("Error while importing wallet from output descriptor: {Error}", addResult.Item2);
+                    return (false, addResult.Item2);
+                }
+                
+                //Track wallet
+                var derivationStrategyBase = wallet.GetDerivationStrategy();
+                if (derivationStrategyBase == null)
+                {
+                    _logger.LogError("Error while getting the derivation scheme");
+                    return (false, "Error while getting the derivation scheme");
+                }
+
+
+                await TrackAndScanWallet(derivationStrategyBase);
+
+
+            }
+            catch (Exception e)
+            {
+                var errorWhileImportingWatchOnlyWalletFromOutputDescriptor = "Error while importing watch-only wallet from output descriptor";
```

Comment:
is there any need for this variable to have such as long name? xD

Reply:
Its automatically done by rider 🫶